java/bundles/org.eclipse.set.feature.plazmodel/src/org/eclipse/set/feature/plazmodel/check/GeoCoordinateValid.java

```diff
@@ -0,0 +1,223 @@
+			});
+		}));
+		return result;
+	}
+
+	private PlazError creatErrorReport(final GEO_Punkt gp) {
+		EObject errObj = gp;
+		if (gp.getGEOPunktAllg() == null) {
+			errObj = gp.getGEOPunktAllg();
+		} else if (gp.getGEOPunktAllg().getGKX() == null) {
+			errObj = gp.getGEOPunktAllg().getGKX();
+		} else if (gp.getGEOPunktAllg().getGKY() == null) {
+			errObj = gp.getGEOPunktAllg().getGKY();
+		}
+		return createGeoCoordinateError(errObj,
+				"GEO_Punkt: {GUID} fehlt Koordinate",
+				Map.of(gp.getIdentitaet().getWert(), "GUID"));
+	}
+
+	private PlazError createErrorReport(final Punkt_Objekt po,
+			final Punkt_Objekt_TOP_Kante_AttributeGroup potk,
+			final GEO_Punkt gp, final double diff) {
+		final ID_GEO_Punkt_ohne_Proxy_TypeClass errorObject = potk
+				.getIDGEOPunktBerechnet().stream()
+				.filter(idgp -> idgp.getValue().equals(gp)).findFirst()
+				.orElse(null);
+		if (errorObject == null) {
+			throw new IllegalArgumentException(String.format(
+					"GEO_Punkt: %s gehört nicht zu Punkt_Objekt: %s",
+					gp.getIdentitaet().getWert(),
+					po.getIdentitaet().getWert()));
+		}
+		return createGeoCoordinateError(errorObject,
+				"Die Koordinaten des GEO_Punkt: {GEO_PUNKT_ID}, der auf die Punkt_Objekt: {GUID} verweisen würde, weicht {DIFF}cm von der berechneten Koordinate ab.",
+				Map.of("GEO_PUNKT_ID", errorObject.getWert(), "GUID",
+						po.getIdentitaet().getWert(), "DIFF",
+						new DecimalFormat("0.00").format(diff * 100)));
+	}
+
+	private PlazError createGeoCoordinateError(final EObject object,
+			final String message, final Map<String, String> data) {
+		final PlazError plazError = PlazFactory.eINSTANCE.createPlazError();
+		plazError.setObject(object);
+		plazError.setSeverity(ValidationSeverity.ERROR);
+		plazError.setType(checkType());
+		plazError
+				.setMessage(StringSubstitutor.replace(message, data, "{", "}"));
+		return plazError;
+	}
+
+	@Override
+	public String checkType() {
+		return "Koordinate";
+	}
+
+	@Override
+	public String getDescription() {
+		return "Die Verweise vom Punkt_Objekt auf den Geo_Punkt durch ID_Geo_Punkt_Berechnen sind plausibilisiert";
+	}
+
+	@Override
+	public String getGeneralErrMsg() {
+		return "Die Verweise vom Punkt_Objekt auf den Geo_Punkt durch ID_Geo_Punkt_Berechnen sind nicht plausibilisiert";
+	}
+
+	private PlazError createProcessingWarning() {
+		final PlazError plazError = PlazFactory.eINSTANCE.createPlazError();
+		plazError.setType(checkType());
+		plazError.setSeverity(ValidationSeverity.WARNING);
+		plazError.setMessage(
+				"Die Prüfung auf ID_GEO_Punkt_Berechnen is noch nicht beenden");
+		return plazError;
+	}
+
+	/**
+	 * @param potk
+	 *            the {@link Punkt_Objekt_TOP_Kante_AttributeGroup}
+	 * @return true, when exist predetermined coordinates through
+	 *         {@link Punkt_Objekt_TOP_Kante_AttributeGroup#getIDGEOPunktBerechnet}
+	 *         and the coordinates haven't same coordinate system
+	 */
+	public static boolean isNotDistincCoordinateSystem(
```

Comment:
@TruongQuangSB Typo in the function name

```suggestion
	public static boolean isNotDistinctCoordinateSystem(
```